\doxysection{pip.\+\_\+internal.\+exceptions.\+Hash\+Missing Class Reference}
\hypertarget{classpip_1_1__internal_1_1exceptions_1_1HashMissing}{}\label{classpip_1_1__internal_1_1exceptions_1_1HashMissing}\index{pip.\_internal.exceptions.HashMissing@{pip.\_internal.exceptions.HashMissing}}


Inheritance diagram for pip.\+\_\+internal.\+exceptions.\+Hash\+Missing\+:
% FIG 0


Collaboration diagram for pip.\+\_\+internal.\+exceptions.\+Hash\+Missing\+:
% FIG 1
\doxysubsubsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
None \mbox{\hyperlink{classpip_1_1__internal_1_1exceptions_1_1HashMissing_abe445b2414fd40af118661495d9b5a77}{\+\_\+\+\_\+init\+\_\+\+\_\+}} (self, str gotten\+\_\+hash)
\item 
str \mbox{\hyperlink{classpip_1_1__internal_1_1exceptions_1_1HashMissing_afe037f000b99010da015d0209089d6dd}{body}} (self)
\end{DoxyCompactItemize}
\doxysubsection*{Public Member Functions inherited from \mbox{\hyperlink{classpip_1_1__internal_1_1exceptions_1_1HashError}{pip.\+\_\+internal.\+exceptions.\+Hash\+Error}}}
\begin{DoxyCompactItemize}
\item 
str \mbox{\hyperlink{classpip_1_1__internal_1_1exceptions_1_1HashError_a455e73a1648e11e2d124cf5bd30f97f3}{\+\_\+\+\_\+str\+\_\+\+\_\+}} (self)
\end{DoxyCompactItemize}
\doxysubsubsection*{Public Attributes}
\begin{DoxyCompactItemize}
\item 
\mbox{\hyperlink{classpip_1_1__internal_1_1exceptions_1_1HashMissing_ac0609ac6fee8ef5fc670a982f6a6eeaa}{gotten\+\_\+hash}} = gotten\+\_\+hash
\end{DoxyCompactItemize}
\doxysubsubsection*{Additional Inherited Members}
\doxysubsection*{Static Public Attributes inherited from \mbox{\hyperlink{classpip_1_1__internal_1_1exceptions_1_1HashError}{pip.\+\_\+internal.\+exceptions.\+Hash\+Error}}}
\begin{DoxyCompactItemize}
\item 
Optional \mbox{\hyperlink{classpip_1_1__internal_1_1exceptions_1_1HashError_af3c50ea4f7da6636a64b938cda7052c9}{req}} = None
\item 
str \mbox{\hyperlink{classpip_1_1__internal_1_1exceptions_1_1HashError_ad051d7f7e98b51e235a5bc4ca17b302c}{head}} = "{}"{}
\item 
int \mbox{\hyperlink{classpip_1_1__internal_1_1exceptions_1_1HashError_a81cbe1b130a39b1625d11456709e1047}{order}} = -\/1
\end{DoxyCompactItemize}
\doxysubsection*{Protected Member Functions inherited from \mbox{\hyperlink{classpip_1_1__internal_1_1exceptions_1_1HashError}{pip.\+\_\+internal.\+exceptions.\+Hash\+Error}}}
\begin{DoxyCompactItemize}
\item 
str \mbox{\hyperlink{classpip_1_1__internal_1_1exceptions_1_1HashError_a17be3100accef085731b0299b3528866}{\+\_\+requirement\+\_\+name}} (self)
\end{DoxyCompactItemize}


\doxysubsection{Detailed Description}
\begin{DoxyVerb}A hash was needed for a requirement but is absent.\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{pip_2__internal_2exceptions_8py_source_l00512}{512}} of file \mbox{\hyperlink{pip_2__internal_2exceptions_8py_source}{exceptions.\+py}}.



\doxysubsection{Constructor \& Destructor Documentation}
\Hypertarget{classpip_1_1__internal_1_1exceptions_1_1HashMissing_abe445b2414fd40af118661495d9b5a77}\index{pip.\_internal.exceptions.HashMissing@{pip.\_internal.exceptions.HashMissing}!\_\_init\_\_@{\_\_init\_\_}}
\index{\_\_init\_\_@{\_\_init\_\_}!pip.\_internal.exceptions.HashMissing@{pip.\_internal.exceptions.HashMissing}}
\doxysubsubsection{\texorpdfstring{\_\_init\_\_()}{\_\_init\_\_()}}
{\footnotesize\ttfamily \label{classpip_1_1__internal_1_1exceptions_1_1HashMissing_abe445b2414fd40af118661495d9b5a77} 
 None pip.\+\_\+internal.\+exceptions.\+Hash\+Missing.\+\_\+\+\_\+init\+\_\+\+\_\+ (\begin{DoxyParamCaption}\item[{}]{self}{, }\item[{str}]{gotten\+\_\+hash}{}\end{DoxyParamCaption})}

\begin{DoxyVerb}:param gotten_hash: The hash of the (possibly malicious) archive we
    just downloaded
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{pip_2__internal_2exceptions_8py_source_l00526}{526}} of file \mbox{\hyperlink{pip_2__internal_2exceptions_8py_source}{exceptions.\+py}}.



Referenced by \mbox{\hyperlink{kernels_8py_source_l00178}{sklearn.\+gaussian\+\_\+process.\+kernels.\+Kernel.\+get\+\_\+params()}}, and \mbox{\hyperlink{__distn__infrastructure_8py_source_l00643}{scipy.\+stats.\+\_\+distn\+\_\+infrastructure.\+rv\+\_\+generic.\+random\+\_\+state()}}.



\doxysubsection{Member Function Documentation}
\Hypertarget{classpip_1_1__internal_1_1exceptions_1_1HashMissing_afe037f000b99010da015d0209089d6dd}\index{pip.\_internal.exceptions.HashMissing@{pip.\_internal.exceptions.HashMissing}!body@{body}}
\index{body@{body}!pip.\_internal.exceptions.HashMissing@{pip.\_internal.exceptions.HashMissing}}
\doxysubsubsection{\texorpdfstring{body()}{body()}}
{\footnotesize\ttfamily \label{classpip_1_1__internal_1_1exceptions_1_1HashMissing_afe037f000b99010da015d0209089d6dd} 
 str pip.\+\_\+internal.\+exceptions.\+Hash\+Missing.\+body (\begin{DoxyParamCaption}\item[{}]{self}{}\end{DoxyParamCaption})}

\begin{DoxyVerb}Return a summary of me for display under the heading.

This default implementation simply prints a description of the
triggering requirement.

:param req: The InstallRequirement that provoked this error, with
    its link already populated by the resolver's _populate_link().
\end{DoxyVerb}
 

Reimplemented from \mbox{\hyperlink{classpip_1_1__internal_1_1exceptions_1_1HashError_a8b1a0890122218544dd1a3e23c416431}{pip.\+\_\+internal.\+exceptions.\+Hash\+Error}}.



Definition at line \mbox{\hyperlink{pip_2__internal_2exceptions_8py_source_l00533}{533}} of file \mbox{\hyperlink{pip_2__internal_2exceptions_8py_source}{exceptions.\+py}}.



\doxysubsection{Member Data Documentation}
\Hypertarget{classpip_1_1__internal_1_1exceptions_1_1HashMissing_ac0609ac6fee8ef5fc670a982f6a6eeaa}\index{pip.\_internal.exceptions.HashMissing@{pip.\_internal.exceptions.HashMissing}!gotten\_hash@{gotten\_hash}}
\index{gotten\_hash@{gotten\_hash}!pip.\_internal.exceptions.HashMissing@{pip.\_internal.exceptions.HashMissing}}
\doxysubsubsection{\texorpdfstring{gotten\_hash}{gotten\_hash}}
{\footnotesize\ttfamily \label{classpip_1_1__internal_1_1exceptions_1_1HashMissing_ac0609ac6fee8ef5fc670a982f6a6eeaa} 
pip.\+\_\+internal.\+exceptions.\+Hash\+Missing.\+gotten\+\_\+hash = gotten\+\_\+hash}



Definition at line \mbox{\hyperlink{pip_2__internal_2exceptions_8py_source_l00531}{531}} of file \mbox{\hyperlink{pip_2__internal_2exceptions_8py_source}{exceptions.\+py}}.



The documentation for this class was generated from the following file\+:\begin{DoxyCompactItemize}
\item 
/home/jam/\+Research/\+IRES-\/2025/dev/src/llm-\/scripts/testing/hypothesis-\/testing/hyp-\/env/lib/python3.\+12/site-\/packages/pip/\+\_\+internal/exceptions.\+py\end{DoxyCompactItemize}
